src/core/server/plugins/discovery/plugins_discovery.ts

```diff
@@ -105,6 +106,34 @@ function processPluginSearchPaths$(pluginDirs: ReadonlyArray<string>, log: Logge
   );
 }
 
+function readSchemaMaybe(
+  pluginPath: string,
+  manifest: PluginManifest,
+  log: Logger
+): PluginConfigSchema {
+  if (!manifest.server) return null;
+  const pluginPathServer = join(pluginPath, 'server');
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const pluginDefinition = require(pluginPathServer);
+
+  if (!('configDefinition' in pluginDefinition)) {
+    log.debug(`"${pluginPathServer}" does not export "configDefinition".`);
```

Comment:
__note:__ there is a chance that this going to be a bit noisy especially when we have more UI-only plugins etc, not sure how useful it's at post-development time. But up to you.

Reply:
there is a check 
```js
if (!manifest.server) return null;
``` 
I'm afraid it's easy to break a convention (with a typo, for example) and would be nice to log a message at least

Reply:
Ah, missed that. Sure, if you feel it's useful then I'm fine